### ISSUE TITLE
Error "Unexpected result from listfunds" in log running CLN 23.05

### ISSUE THREAD

**Issue by kallerosenbaum** [ksedgwic/clboss#164]
I run clboss 0.13A and I upgraded my CLN node from 23.02.2 to 23.05 and my logfile looks something like this

```
2023-05-14T09:36:40.842Z INFO    plugin-bcli: bitcoin-cli initialized and connected to bitcoind.
2023-05-14T09:37:39.618Z INFO    plugin-clboss: clboss 0.13A
2023-05-14T09:37:39.618Z INFO    plugin-clboss: ChannelFeeManager: zerobasefee: require
2023-05-14T09:37:42.269Z INFO    plugin-clboss: DnsSeed: Have 2 seeds.
2023-05-14T09:37:42.269Z INFO    plugin-clboss: Started.
2023-05-14T09:37:42.314Z INFO    plugin-clboss: New block at 789649
2023-05-14T09:37:42.323Z **BROKEN** plugin-clboss: ListfundsAnalyzer: Unexpected result from listfunds: outputs = [\n\t{\n\t\t\"txid\" : \"....

...

2023-05-14T09:37:42.370Z **BROKEN** plugin-clboss: NodeBalanceSwapper: Unexpected listpeers: [\n\t{\n\t\t\"id\" : \"

...

2023-05-14T09:37:43.419Z **BROKEN** plugin-clboss: PeerCompetitorFeeMonitor: unexpected listchannels result: {\n\t\"source\" : \"

...
```

It seems CLN made some changes to their RPC interface causing a lot of error logs from clboss. However, I also upgraded clboss from 0.12 at the same time, so I can't really tell which change causes the issue. I assume it's the upgrade of CLN since 0.13A is relatively old.

**Comment by kallerosenbaum** [comment]
Wild guess: Maybe this https://github.com/ElementsProject/lightning/releases/tag/v23.05:

```
Removed msat suffix from all millisatoshi fields, as deprecated in v0.12.0.
```

**Comment by ksedgwic** [comment]
I believe that https://github.com/ZmnSCPxj/clboss/pull/162 fixes this

**Comment by xNDTyf** [comment]
@ksedgwic Won't you make a fork under your account & merge all your improvements? It looks that the author is no more active ;)

**Comment by ksedgwic** [comment]
I'm thinking about it.  I think there is a good chance I'll know more in early July.

**Comment by k0gen** [comment]
> I'm thinking about it. I think there is a good chance I'll know more in early July.

Great job, @ksedgwic ! All CLBOOS users are going to love you for it! 🙏

**Comment by JaviLib** [comment]
> I'm thinking about it. I think there is a good chance I'll know more in early July.

Please do it, I think you have the support of the community.

**Comment by k0gen** [comment]
@ZmnSCPxj good to see you are back!

**Comment by JaviLib** [comment]
Is this solved after commit b2def4c51e78904e1021bfc2b3bd74e2ec7515a9 ?

**Comment by martinneustein** [comment]
I can't compile on Raspberry Pi or Odroid (same error):

```

Boss/Mod/JitRebalancer.cpp: In lambda function:
Boss/Mod/JitRebalancer.cpp:137:56: error: cannot convert ‘const CommandId’ {aka ‘const Util::Either<long unsigned int, std::__cxx11::basic_string<char> >’} to ‘uint64_t’ {aka ‘long unsigned int’}
  137 |                                                  , req.id
      |                                                    ~~~~^~
      |                                                        |
      |                                                        const CommandId {aka const Util::Either<long unsigned int, std::__cxx11::basic_string<char> >}
Boss/Mod/JitRebalancer.cpp:144:43: note:   initializing argument 2 of ‘Ev::Io<bool> Boss::Mod::JitRebalancer::Impl::htlc_accepted_cont(const Ln::NodeId&, uint64_t, Ln::Amount)’
  144 |                           , std::uint64_t id
      |                             ~~~~~~~~~~~~~~^~
Boss/Mod/JitRebalancer.cpp: In lambda function:
Boss/Mod/JitRebalancer.cpp:157:65: error: invalid initialization of reference of type ‘const CommandId&’ {aka ‘const Util::Either<long unsigned int, std::__cxx11::basic_string<char> >&’} from expression of type ‘const uint64_t’ {aka ‘const long unsigned int’}
  157 |                                                 , stringify_cid(id).c_str()
      |                                                                 ^~
Boss/Mod/JitRebalancer.cpp:72:48: note: in passing argument 1 of ‘std::string {anonymous}::stringify_cid(const CommandId&)’
   72 | std::string stringify_cid(Ln::CommandId const& id) {
      |                           ~~~~~~~~~~~~~~~~~~~~~^~
Boss/Mod/JitRebalancer.cpp:163:79: error: cannot convert ‘const uint64_t’ {aka ‘const long unsigned int’} to ‘Ln::CommandId’ {aka ‘Util::Either<long unsigned int, std::__cxx11::basic_string<char> >’}
  163 |                         return Boss::concurrent( check_and_move(node, amount, id)
      |                                                                               ^~
      |                                                                               |
      |                                                                               const uint64_t {aka const long unsigned int}
Boss/Mod/JitRebalancer.cpp:196:39: note:   initializing argument 3 of ‘Ev::Io<void> Boss::Mod::JitRebalancer::Impl::check_and_move(const Ln::NodeId&, Ln::Amount, Ln::CommandId)’
  196 |                       , Ln::CommandId id
      |                         ~~~~~~~~~~~~~~^~
Boss/Mod/JitRebalancer.cpp: In member function ‘Ev::Io<bool> Boss::Mod::JitRebalancer::Impl::htlc_accepted_cont(const Ln::NodeId&, uint64_t, Ln::Amount)’:
Boss/Mod/JitRebalancer.cpp:147:54: error: no matching function for call to ‘Ev::Io<const std::set<Ln::NodeId>*>::then(Boss::Mod::JitRebalancer::Impl::htlc_accepted_cont(const Ln::NodeId&, uint64_t, Ln::Amount)::<lambda(const std::set<Ln::NodeId>*)>)’
  147 |                 return unmanager.get_unmanaged().then([ this
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
  148 |                                                       , node
      |                                                       ~~~~~~
  149 |                                                       , amount
      |                                                       ~~~~~~~~
  150 |                                                       , id
      |                                                       ~~~~
  151 |                                                       ](std::set<Ln::NodeId> const* unmanaged) {
      |                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  152 |                         if (unmanaged->count(node) != 0) {
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  153 |                                 return Boss::log( bus, Debug
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  154 |                                                 , "JitRebalancer: HTLC %s to "
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  155 |                                                   "unmanaged node %s, will "
      |                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~
  156 |                                                   "ignore."
      |                                                   ~~~~~~~~~
  157 |                                                 , stringify_cid(id).c_str()
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  158 |                                                 , Util::stringify(node).c_str()
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  159 |                                                 ).then([]() {
      |                                                 ~~~~~~~~~~~~~
  160 |                                         return Ev::lift(false);
      |                                         ~~~~~~~~~~~~~~~~~~~~~~~
  161 |                                 });
      |                                 ~~~
  162 |                         }
      |                         ~
  163 |                         return Boss::concurrent( check_and_move(node, amount, id)
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  164 |                                                ).then([]() {
      |                                                ~~~~~~~~~~~~~
  165 |                                 return Ev::lift(true);
      |                                 ~~~~~~~~~~~~~~~~~~~~~~
  166 |                         });
      |                         ~~~
  167 |                 });
      |                 ~~
In file included from ./Boss/ModG/ReqResp.hpp:7,
                 from ./Boss/ModG/RebalanceUnmanagerProxy.hpp:4,
                 from Boss/Mod/JitRebalancer.cpp:2:
./Ev/Io.hpp:95:9: note: candidate: ‘template<class f> Ev::Io<typename Ev::Detail::IoInner<typename std::result_of<f(a)>::type>::type> Ev::Io<a>::then(f) && [with f = f; a = const std::set<Ln::NodeId>*]’
   95 |         then(f func)&& {
      |         ^~~~
./Ev/Io.hpp:95:9: note:   template argument deduction/substitution failed:
./Ev/Io.hpp: In substitution of ‘template<class f> Ev::Io<typename Ev::Detail::IoInner<typename std::result_of<f(const std::set<Ln::NodeId>*)>::type>::type> Ev::Io<const std::set<Ln::NodeId>*>::then<f>(f) && [with f = Boss::Mod::JitRebalancer::Impl::htlc_accepted_cont(const Ln::NodeId&, uint64_t, Ln::Amount)::<lambda(const std::set<Ln::NodeId>*)>]’:
Boss/Mod/JitRebalancer.cpp:147:40:   required from here
./Ev/Io.hpp:95:9: error: no type named ‘type’ in ‘struct std::result_of<Boss::Mod::JitRebalancer::Impl::htlc_accepted_cont(const Ln::NodeId&, uint64_t, Ln::Amount)::<lambda(const std::set<Ln::NodeId>*)>(const std::set<Ln::NodeId>*)>’
mv -f Boss/Mod/.deps/libclboss_la-JsonOutputter.Tpo Boss/Mod/.deps/libclboss_la-JsonOutputter.Plo
/bin/bash ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.  -I./external/basicsecure/ -I./external/bitcoin-ripemd160/ -I./external/bitcoin-sha256/ -I./external/secp256k1/include/ -I./external/jsmn  -Wall -Werror -pthread    -Wno-noexcept-type  -O2 -MT Boss/Mod/libclboss_la-ListfundsAnnouncer.lo -MD -MP -MF Boss/Mod/.deps/libclboss_la-ListfundsAnnouncer.Tpo -c -o Boss/Mod/libclboss_la-ListfundsAnnouncer.lo `test -f 'Boss/Mod/ListfundsAnnouncer.cpp' || echo './'`Boss/Mod/ListfundsAnnouncer.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I./external/basicsecure/ -I./external/bitcoin-ripemd160/ -I./external/bitcoin-sha256/ -I./external/secp256k1/include/ -I./external/jsmn -Wall -Werror -pthread -Wno-noexcept-type -O2 -MT Boss/Mod/libclboss_la-ListfundsAnnouncer.lo -MD -MP -MF Boss/Mod/.deps/libclboss_la-ListfundsAnnouncer.Tpo -c Boss/Mod/ListfundsAnnouncer.cpp -o Boss/Mod/libclboss_la-ListfundsAnnouncer.o
mv -f Boss/Mod/.deps/libclboss_la-InternetConnectionMonitor.Tpo Boss/Mod/.deps/libclboss_la-InternetConnectionMonitor.Plo
/bin/bash ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.  -I./external/basicsecure/ -I./external/bitcoin-ripemd160/ -I./external/bitcoin-sha256/ -I./external/secp256k1/include/ -I./external/jsmn  -Wall -Werror -pthread    -Wno-noexcept-type  -O2 -MT Boss/Mod/libclboss_la-ListpaysHandler.lo -MD -MP -MF Boss/Mod/.deps/libclboss_la-ListpaysHandler.Tpo -c -o Boss/Mod/libclboss_la-ListpaysHandler.lo `test -f 'Boss/Mod/ListpaysHandler.cpp' || echo './'`Boss/Mod/ListpaysHandler.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I./external/basicsecure/ -I./external/bitcoin-ripemd160/ -I./external/bitcoin-sha256/ -I./external/secp256k1/include/ -I./external/jsmn -Wall -Werror -pthread -Wno-noexcept-type -O2 -MT Boss/Mod/libclboss_la-ListpaysHandler.lo -MD -MP -MF Boss/Mod/.deps/libclboss_la-ListpaysHandler.Tpo -c Boss/Mod/ListpaysHandler.cpp -o Boss/Mod/libclboss_la-ListpaysHandler.o
mv -f Boss/Mod/.deps/libclboss_la-ListfundsAnalyzer.Tpo Boss/Mod/.deps/libclboss_la-ListfundsAnalyzer.Plo
/bin/bash ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.  -I./external/basicsecure/ -I./external/bitcoin-ripemd160/ -I./external/bitcoin-sha256/ -I./external/secp256k1/include/ -I./external/jsmn  -Wall -Werror -pthread    -Wno-noexcept-type  -O2 -MT Boss/Mod/libclboss_la-ListpeersAnalyzer.lo -MD -MP -MF Boss/Mod/.deps/libclboss_la-ListpeersAnalyzer.Tpo -c -o Boss/Mod/libclboss_la-ListpeersAnalyzer.lo `test -f 'Boss/Mod/ListpeersAnalyzer.cpp' || echo './'`Boss/Mod/ListpeersAnalyzer.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I./external/basicsecure/ -I./external/bitcoin-ripemd160/ -I./external/bitcoin-sha256/ -I./external/secp256k1/include/ -I./external/jsmn -Wall -Werror -pthread -Wno-noexcept-type -O2 -MT Boss/Mod/libclboss_la-ListpeersAnalyzer.lo -MD -MP -MF Boss/Mod/.deps/libclboss_la-ListpeersAnalyzer.Tpo -c Boss/Mod/ListpeersAnalyzer.cpp -o Boss/Mod/libclboss_la-ListpeersAnalyzer.o
make[2]: *** [Makefile:4607: Boss/Mod/libclboss_la-JitRebalancer.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
mv -f Boss/Mod/.deps/libclboss_la-ListfundsAnnouncer.Tpo Boss/Mod/.deps/libclboss_la-ListfundsAnnouncer.Plo
mv -f Boss/Mod/.deps/libclboss_la-ListpeersAnalyzer.Tpo Boss/Mod/.deps/libclboss_la-ListpeersAnalyzer.Plo
mv -f Boss/Mod/.deps/libclboss_la-ListpaysHandler.Tpo Boss/Mod/.deps/libclboss_la-ListpaysHandler.Plo
make[2]: Leaving directory '/home/admin/tmp/clboss'
make[1]: *** [Makefile:5469: all-recursive] Error 1
make[1]: Leaving directory '/home/admin/tmp/clboss'
make: *** [Makefile:2369: all] Error 2
```

**Comment by tsjk** [comment]
Can this also be closed now?

**Comment by ksedgwic** [comment]
Fixed by #162
